### PR TITLE
Remove cache steps from build workflow

### LINKED ISSUE #225: Clear stale cache artifacts built against old Rust versions
Prompted from build issues in PR #220 - @zbraniecki 

Initial idea: if the compatibility of cached build artifacts to the current build also depend on Rust versions matching, then let's add the Rust version to the cache key.

To test: does running the `rustc` command on the Github Actions runner VMs automatically access the latest stable Rust, or do we need to run a rust command (`rustup` via [actions-rs/toolchain](https://github.com/actions-rs/toolchain)?) to get the latest Rust?

### PR DESCRIPTION
**Pull request by echeran**
Closes #225

### CODE REVIEW

**Comment by echeran** [comment]
@zbraniecki Here is the change to remove caching for now, as discussed in the issue.  It seems to work -- your PR now gets past build issues and [produces a different error](https://github.com/unicode-org/icu4x/pull/236/checks?check_run_id=1103871582#step:4:207).  I created a testing PR (#236) to test it out the effect on your PR.  I added a TODO + new issue for later for exploring re-enabling caching.

**Review by sffc — APPROVED** [review]

**Comment by sffc** [comment]
Waiting for a review from @zbraniecki

**Review by zbraniecki — APPROVED** [review]

**Pull request merged**
